### PR TITLE
Use dynamic project data

### LINKED ISSUE #22: Set up a project config
The project title (_Science Gossip_), name (_illustratedlife_) and resource prefix(_SC_) are embedded into the code in a few places. They'll need to be extracted out into a data file so that we can run the build across different projects.

Eleventy looks in `_data` for global data so a `project.json` file in that directory should set up a global `project` variable that's available to all templates.

The regex that matches subject mentions will also need to be generalised here:
https://github.com/zooniverse/Talk-archiver/blob/23ddb12b8b6860f9a8e952c5d226c8477c75f7d8/.eleventy.js#L175-L183

### PR DESCRIPTION
**Pull request by eatyourgreens**
- Get project data from the Ouroboros export files during the data build.
- Set project data as a global data object during the site build. 
- Build subject and collection mention regexes dynamically from the project prefix, when the eleventy config runs.
- Build paths from the project name, when reading data exports.
- Store the project name and prefix, needed to start a build, in one place for convenient editing. Edit `src/config/index.js` to change project.

Closes #22.

### CODE REVIEW

**Comment by camallen** [comment]
fwiw this data file should exist in the the converted backups archive, `projects.json`. Maybe not formatted correctly but i did include it.

**Comment by eatyourgreens** [comment]
Cool. Turns out Eleventy doesn't support async functions in `.eleventy.js` so, at the moment, I can't read in the project config from the API or a file.

**Comment by eatyourgreens** [comment]
@camallen I've added the `projects` file in 9f48c12.

**Comment by eatyourgreens** [comment]
Still to do: 
- [ ] build the data file names dynamically, from the project name, when reading the exports.
- [ ] consolidate the project settings (name and file prefix) in one place so they can be edited together.

**Comment by eatyourgreens** [comment]
Ok, this should be ready to go now.

File: src/config/index.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  project: {
+    name: 'illustratedlife',
```
Comment (camallen) [inline src/config/index.js:3]: Is the plan to override this for each talk project we want to archive? Should we use ENV vars in here to inject via the build scripts?

**Comment by eatyourgreens** [comment]
This runs out of memory when I run it with P4 so I'm going to hold off on merging it.

File: src/config/index.js (hunk 1 above)
Comment (eatyourgreens) [reply src/config/index.js:3]: The deploy target in `package.json` will need to be changed too.

**Comment by eatyourgreens** [comment]
With 12GB allocated to Node, this still crashes when building the site data for Planet Four (~80k resources.) I think that might need looking at in a separate PR.

```
building tag manymanymanyfans
building tag double-smoke

<--- Last few GCs --->

[16958:0x104013000]   810431 ms: Mark-sweep 11869.4 (12032.6) -> 11861.7 (12032.1) MB, 23793.8 / 0.0 ms  (average mu = 0.131, current mu = 0.013) allocation failure scavenge might not succeed
[16958:0x104013000]   836343 ms: Mark-sweep 11873.5 (12040.3) -> 11869.9 (12040.3) MB, 25748.8 / 0.0 ms  (average mu = 0.070, current mu = 0.006) allocation failure scavenge might not succeed
```

File: src/config/index.js (hunk 1 above)
Comment (eatyourgreens) [reply src/config/index.js:3]: I think you're right that the final version will get site name, prefix and deploy target from ENV.

**Pull request merged**